flow/connectors/mysql/qrep.go

```diff
@@ -224,26 +218,59 @@ func (c *MySqlConnector) PullQRepRecords(
 
 	if last.FullTablePartition {
 		// this is a full table partition, so just run the query
-		if err := c.ExecuteSelectStreaming(ctx, query, &rs, onRow, onResult); err != nil {
+		if err := c.ExecuteSelectStreaming(ctx, config.Query, &rs, onRow, onResult); err != nil {
 			return 0, 0, err
 		}
 	} else {
-		var rangeStart any
-		var rangeEnd any
+		var rangeStart string
+		var rangeEnd string
 
 		// Depending on the type of the range, convert the range into the correct type
 		switch x := last.Range.Range.(type) {
 		case *protos.PartitionRange_IntRange:
-			rangeStart = x.IntRange.Start
-			rangeEnd = x.IntRange.End
+			var unsigned bool
+			for _, col := range tableSchema.Columns {
+				if col.Name == config.WatermarkColumn {
+					switch qvalue.QValueKind(col.Type) {
+					case qvalue.QValueKindUInt8:
+						rangeStart = strconv.FormatUint(uint64(uint8(x.IntRange.Start)), 10)
+						rangeEnd = strconv.FormatUint(uint64(uint8(x.IntRange.End)), 10)
+						unsigned = true
+					case qvalue.QValueKindUInt16:
+						rangeStart = strconv.FormatUint(uint64(uint16(x.IntRange.Start)), 10)
+						rangeEnd = strconv.FormatUint(uint64(uint16(x.IntRange.End)), 10)
+						unsigned = true
+					case qvalue.QValueKindUInt32:
+						rangeStart = strconv.FormatUint(uint64(uint32(x.IntRange.Start)), 10)
+						rangeEnd = strconv.FormatUint(uint64(uint32(x.IntRange.End)), 10)
+						unsigned = true
+					case qvalue.QValueKindUInt64:
+						rangeStart = strconv.FormatUint(uint64(x.IntRange.Start), 10)
+						rangeEnd = strconv.FormatUint(uint64(x.IntRange.End), 10)
+						unsigned = true
```

Comment:
default should error?

Reply:
default is signed integers, we let existing logic handle that same as postgres